### PR TITLE
Doc: Add Rust integrations guide

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

As suggested by @chuck-alt-delete [here](https://materializeinc.slack.com/archives/CPNFV9CVB/p1689963971373969), adding a Rust integrations guide.

### Tips for reviewer

The Rust guide uses the exact same structure as all of the other languages and borrows the code examples from the [connection examples repo](https://github.com/MaterializeInc/connection-examples/tree/main)

### CODE REVIEW

**Review by chuck-alt-delete — CHANGES_REQUESTED** [review]
Ideally this would be something someone can follow along and run. To make this flow for someone following along, I would suggest first creating the source, then creating the countries table and add a row every second in a loop, then creating a view, then selecting from it, then subscribing to the view and seeing the changes. Include cargo run commands for the insert and subscribe programs in separate terminals, and remind folks to Ctrl+C to stop them.

What do you think?

**Comment by bobbyiliev** [comment]
> Ideally this would be something someone can follow along and run. To make this flow for someone following along, I would suggest first creating the source, then creating the countries table and add a row every second in a loop, then creating a view, then selecting from it, then subscribing to the view and seeing the changes. Include cargo run commands for the insert and subscribe programs in separate terminals, and remind folks to Ctrl+C to stop them.
> 
> What do you think?

I like the idea of this being something that people can follow along! But I think that having a loop and asking the users to run this in separate terminal windows might be confusing. We could just use the load generator source for the subscribe example instead of hammering environmentd with inserts in a loop.

I'll restructure this so that the flow actually makes sense!

File: doc/user/content/integrations/rust.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    let results = client.query("SELECT code, name FROM countries;", &[]).expect("Error running query.");
+
+    for row in results {
+        println!("{:} - {:}", row.get::<usize, String>(0), row.get::<usize, String>(1));
+    };
+}
+```
+
+## Insert data into tables
```
[the remark this answers is not in the document]
Comment (chuck-alt-delete) [reply doc/user/content/integrations/rust.md:57]: That’s awesome!

we should probably also have a teardown section to drop the table and the source.

If it’s not too much trouble, having this improved flow consistent across examples would be great

**Review by chuck-alt-delete — APPROVED** [review]

File: doc/user/content/integrations/rust.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+description: "Use Rust tokio-postgres to connect, insert, manage, query and stream from Materialize."
+aliases:
+  - /guides/rust
+menu:
+  main:
+    parent: 'client-libraries'
+    name: "Rust"
+---
+
+Materialize is **wire-compatible** with PostgreSQL, which means that Rust applications can use common PostgreSQL clients to interact with Materialize. In this guide, we'll use the [`tokio-postgres`](https://docs.rs/tokio-postgres/0.7.2/tokio_postgres/) crate to connect to Materialize and issue SQL commands.
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply doc/user/content/integrations/rust.md:12]: Good point! `postgres-openssl` is the TLS support for [`tokio-postgres`](https://crates.io/crates/tokio-postgres) via `openssl`, I'll add a clarification in there!

**Comment by morsapaes** [comment]
> Ideally this would be something someone can follow along and run.

Can't find it now, but we have an issue somewhere exactly to make **all** these guides runnable, and based on the load generator (or, rather, to use whatever self-contained flow we use for the quickstart). I'd suggest we keep this guide consistent with the others, and make them consistently runnable as a follow-up.

**Review by joacoc — APPROVED** [review]

File: doc/user/content/integrations/_index.md [hunk 3]
```diff
@@ -182,6 +182,7 @@ Client libraries and ORM frameworks tend to run complex introspection queries th
 | PHP      | {{< supportLevel production >}} | [`pdo_pgsql`](https://www.php.net/manual/en/ref.pgsql.php) | See the [PHP cheatsheet](/integrations/php/).         |
 | Python   | {{< supportLevel production >}} | [`psycopg2`](https://pypi.org/project/psycopg2/)           | See the [Python cheatsheet](/integrations/python/).   |
 | Ruby     | {{< supportLevel production >}} | [`pg` gem](https://rubygems.org/gems/pg/)                  | See the [Ruby cheatsheet](/integrations/ruby/).       |
+| Rust     | {{< supportLevel production >}} | [`postgres`](https://crates.io/crates/postgres)            | See the [Rust cheatsheet](/integrations/rust/).       |
```
Comment (morsapaes) [inline doc/user/content/integrations/_index.md:185]: Should this be `postgres-openssl`?

Comment (bobbyiliev) [reply doc/user/content/integrations/_index.md]: Good catch! Just updated it

**Pull request merged**